### ISSUE TITLE
Topmost result in Quick Search should be highlighted as keyboard focus

### ISSUE THREAD

**Issue by NoRi2909** [home-assistant/frontend#29624]
### Checklist

- [x] I have updated to the latest available Home Assistant version.
- [x] I have cleared the cache of my browser.
- [x] I have tried a different browser to see if it is related to my browser.
- [ ] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

As soon as you have entered anything in the search box of Quick Search hitting Return or Enter will open the topmost search result. But this is not indicated by highlighting:

<img width="918" height="412" alt="Image" src="https://github.com/user-attachments/assets/24361a78-a754-4ab7-93e3-145040bb1c77" />

When you press cursor-down once the topmost search result becomes highlighted and will still be opened with Return or Enter.

Only after pressing cursor-down a second time the next search result gets the keyboard focus.

### Describe the behavior you expected

As soon as the user has entered a single character, the topmost search result should get highlighted.
So after typing "a" it should look like this:

<img width="918" height="412" alt="Image" src="https://github.com/user-attachments/assets/7f3724c0-514e-4a80-9a59-43442288b550" />

Now when you want to open the next search result only a single cursor-down plus Return/Enter will work – which would be the expected behavior.

### Steps to reproduce the issue

See above.

### What version of Home Assistant Core has the issue?

2026.2.1

### What was the last working version of Home Assistant Core?

n/a

**Comment by NoRi2909** [comment]
Keep alive.

**Comment by marcinbauer-ohf** [comment]
This makes sense and helps with accessibility. One thing to watch out for is, that when typing the list re-renders and when I tested highlighting the first item every time, there is a flashing artifact that occurs when the list clears and re-renders. Would be great if we could avoid that.

https://github.com/user-attachments/assets/65f4eaf3-df1c-49ff-9596-188a48d71c16
